src/regions/PolygonRegion.js

```diff
@@ -187,19 +191,49 @@ const Model = types
       },
 
       insertPoint(insertIdx, x, y) {
+        const pointCoords = self.control?.getSnappedPoint({
+          x: self.parent.canvasToInternalX(x),
+          y: self.parent.canvasToInternalY(y),
+        });
+
+        if (
+          (
+            self.points[insertIdx - 1]
+            && self.parent.isSamePixel(pointCoords, self.points[insertIdx - 1])
+          )
+          || (
+            self.points[insertIdx]
+            && self.parent.isSamePixel(pointCoords, self.points[insertIdx])
+          )
+        ) {
```

Comment:
no-no, not like this, please. rewrite it with meaningful names and maybe a comment.